### DISCUSSION TITLE
Installing GeNN using requirements.txt on Streamlit

### DISCUSSION THREAD

**Discussion by russelljjarvis** [genn-team/genn#525] [GeNN questions]
I am seeking to attach [this](https://share.streamlit.io/russelljjarvis/potjansvis/main/app.py) adjustable network visualization of the Potjan's model to simulation parameters in the pygenn's Potjans model. Streamlit is in theory very easy to work with as a front end, however, you can only build things using a Python requirements.txt file and or apt-get commands.

Would it be viable to install pygenn on the streamlit server in such a way?

**Comment by neworderofjamie** [comment]
This should be possible by building your own wheel and installing via requirements.txt with a [direct requirement](https://stackoverflow.com/a/65403005) - this should also work with URLs. In order to build a wheel you just need to run ``python setup.py bdist_wheel`` rather than ``python setup.py develop`` when building PyGeNN. The single-threaded CPU backend should work as long as the wheel is compatible. We are hoping to start distributing PyGeNN on pip but there are currently a few [outstanding issues](https://github.com/genn-team/genn/issues/391)

**Reply by russelljjarvis** [reply]
I guess being a single CPU instance without NVIDIA GPU streamlit wouldn't leverage any of Genns strengths.

From this perspective, it might be worth renting a GPU enabled instance on DigitalOcean or AWS.

**Comment by neworderofjamie** [comment]
Yeah, the single-threaded CPU backend can be very competitive for small models (mostly because the code it generates is so simple compilers can do a pretty good job optimising it) but quickly gets slow for anything more than around 1000 neurons.
